doc/holochain_101/src/zome/read_and_write.md

```diff
@@ -1 +1,154 @@
+
+## Data Propogation
+After any chain-modifying functions are called, Holochain will attempt to notify other peers in the network of those actions. Note that it is not a requirement that the device be online or connected to any other networks or nodes for Holochain to be usable! Holochain will announce the new data over the network, by attempting to publish it to peers. As mentioned previously, if the entry is one of a `private` App Entry type, then only the Header will be published, not the entry itself. Publishing of the data is secondary to being able to author data in the first place, so this will be further elaborated in the [entry validation](./entry_validation.md) article.
+
+## Writing Data
+Broadly speaking, writing data is accomplished by writing Zome source code that makes API calls to Holochain to affect the local source chain. One should only modify chain data by calling Holochain API functions, as there is lots of internal logic that it is instrumental that Holochain perform, for each change. When an API function to alter the chain from Zome source code is invoked, a series of steps is performed internally. It is important to know, at least roughly, what those steps are so that you can use Holochain effectively.
+
+Apart from the following three functions, there is only one additional way that data can be written to the local source chain, which is written about in [linking](./linking.md).
+
+### Creating Entries, or "Committing"
+Of course, first and foremost, there is the simple action of creating a new Entry. This is actually known as "committing" an Entry. It is known as "committing" an Entry in Holochain for precisely the reason that once you write it, you can't "unwrite" it, or delete it, without corrupting the integrity of your local source chain. It is there for good, and can only be "marked" as updated or deleted by writing additional entries in the future.
+
+Because the Entry will be permanent in your local source chain, it must first pass validation, in order to be written in the first place. Validation of Entries is core to the distributed data integrity model of Holochain.
+
+Invoking the commit entry API function will not always return "success". The call can fail if the Entry fails to pass validation, or something else goes wrong internally while Holochain is attempting to write the Entry to the local source chain. Like in any code, failure tolerant code is far better, and Zomes should be written to handle possible error cases.
+
+So in general, the process that Holochain follows while trying to write an Entry during a "commit" call is this:
+
+1. Collect data necessary for use in validation (a "validation package")
+2. Check whether the new entry is valid by passing it through the validation callback specified in the Zome for its' given entry type. If this fails, it will exit here and return either a user defined, or system level error message
```

Comment:
```suggestion
2. Checks whether the new entry is valid by passing it through the validation callback specified in the Zome for its given entry type. If this fails, it will exit here and return either a user defined, or system level error message
```

Reply:
Just out of curiosity, how are those user-defined messages defined? Could be useful to write about if it's available already.

Reply:
That should be covered under the Entry Validation article, which we will talk about today.